typescript/sdk/src/token/types.ts

```diff
@@ -309,10 +309,39 @@ export type HypTokenRouterConfigMailboxOptionalBase = z.infer<
   typeof HypTokenRouterConfigMailboxOptionalBaseSchema
 >;
 
-export const HypTokenRouterConfigMailboxOptionalSchema = z.preprocess(
-  preprocessWarpRouteDeployConfig,
-  HypTokenRouterConfigMailboxOptionalBaseSchema,
-);
+export const HypTokenRouterConfigMailboxOptionalSchema = z
+  .preprocess(
+    preprocessWarpRouteDeployConfig,
+    HypTokenRouterConfigMailboxOptionalBaseSchema,
+  ) // Check that each tokenFee.token is the same as config.token
+  .transform((config, ctx) => {
+    if (!isCollateralTokenConfig(config) || !config.tokenFee) return config;
+
+    if (config.tokenFee.token !== config.token) {
```

Comment:
What happens if the deployment is for a native token? shouldn't we check in that case that the fee token is set to the 0 addy?

Reply:
good question.
this is the input schema so it is expected that `token` will be undefined _for both_  `config.token` and `config.tokenFee.token`. in other words, the existing native config does not have address 0, so it is not expected that config.tokenFee.token to be address 0. 

however, after parsing, we will populate fee.token to be address 0. ideally, both `config.token` and `config.tokenFee.token` are set to address 0 for consistency, but i believe it is fine for now.